js/test/ValidateSPV.test.js

```diff
@@ -1,311 +1,175 @@
     it('returns true if header chain is valid', () => {
-      const res = ValidateSPV.validateHeaderChain(OP_RETURN.HEADER_CHAIN);
-      assert.equal(res, BigInt('49134394618239'));
-    });
-
-    it('throws Error("Header bytes not multiple of 80.") if header chain is not divisible by 80', () => {
-      try {
-        ValidateSPV.validateHeaderChain(HEADER_ERR.HEADER_CHAIN_INVALID_LEN);
-        assert(false, 'expected an error');
-      } catch (e) {
-        assert.include(e.message, 'Header bytes not multiple of 80.');
-      }
-    });
-
-    it('throws Error("Header bytes not a valid chain.") if header chain prevHash is invalid', () => {
-      try {
-        ValidateSPV.validateHeaderChain(HEADER_ERR.HEADER_CHAIN_INVALID_PREVHASH);
-        assert(false, 'expected an error');
-      } catch (e) {
-        assert.include(e.message, 'Header bytes not a valid chain.');
+      for (let i = 0; i < validateHeaderChain.length; i += 1) {
+        const res = ValidateSPV.validateHeaderChain(validateHeaderChain[i].input);
+        assert.equal(res, BigInt(validateHeaderChain[i].output));
       }
     });
 
-    it('throws Error("Header does not meet its own difficulty target.) if a header does not meet its target', () => {
-      try {
-        ValidateSPV.validateHeaderChain(HEADER_ERR.HEADER_CHAIN_LOW_WORK);
-        assert(false, 'expected an error');
-      } catch (e) {
-        assert.include(e.message, 'Header does not meet its own difficulty target.');
+    it('throws Error("Header bytes not multiple of 80.") if header chain is not divisible by 80', () => {
+      for (let i = 0; i < validateHeaderChainError.length; i += 1) {
+        try {
+          ValidateSPV.validateHeaderChain(validateHeaderChainError[i].input);
+          assert(false, 'expected an error');
+        } catch (e) {
+          assert.include(e.message, validateHeaderChainError[i].errorMessage);
+        }
       }
     });
   });
 
   describe('#validateHeaderWork', () => {
-    it('returns false on an empty digest', () => {
-      const res = ValidateSPV.validateHeaderWork(EMPTY, 1);
-      assert.isFalse(res);
-    });
-
-    it('returns false if the digest has insufficient work', () => {
-      const res = ValidateSPV.validateHeaderWork(HEADER.VALIDATE_WORK[0], 1);
-      assert.isFalse(res);
-    });
-
-    it('returns true if the digest has sufficient work', () => {
-      const res = ValidateSPV.validateHeaderWork(
-        OP_RETURN.INDEXED_HEADERS[0].DIGEST_BE,
-        BigInt(HEADER.VALIDATE_WORK[1])
-      );
-      assert.isTrue(res);
+    it('returns true if the digest has sufficient work, returns false if insufficient work or empty digest', () => {
+      for (let i = 0; i < validateHeaderWork.length; i += 1) {
+        let t;
+        if (typeof validateHeaderWork[i].target !== 'number') {
+          t = utils.bytesToUint(validateHeaderWork[i].input.target);
+        } else {
+          t = validateHeaderWork[i].target;
+        }
```

Comment:
`t` should be a `BigInt`, not a `number`
```        let t;
        if (typeof validateHeaderWork[i].input.target === 'number') {
          t = BigInt(validateHeaderWork[i].input.target);        
        } else {
          t = utils.bytesToUint(validateHeaderWork[i].input.target);
        }
```

Reply:
the JSDoc for `validateHeaderWork` is wrong. It's in ValidateSPV.js on line 178

 * @param {BigInt}    target The target threshold